design-patterns/behavioral-patterns/src/templateMethod/OrderTemplate.java

```diff
@@ -0,0 +1,32 @@
+package templateMethod;
+
+public abstract class OrderTemplate {
+	
+	public boolean isGift;
+
+	public abstract void doCheckout();
+
+	public abstract void doPayment();
+
+	public abstract void doReceipt();
+
+	public abstract void doDelivery();
```

Comment:
I would suggest some names for these methods that I would choose. Let me know what you think about them.

```
checkout()

pay()

receive()

deliver()


```